src/main/java/org/apache/fineract/utils/CsvUtility.java

```diff
@@ -0,0 +1,52 @@
+package org.apache.fineract.utils;
+
+import org.apache.fineract.exception.WriteToCsvException;
+import org.apache.fineract.operations.TransactionRequest;
+
+import javax.servlet.http.HttpServletResponse;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+
+public class CsvUtility {
```

Comment:
Since this class is not a general-purpose utility and rather more specific to the transactionRequest type, The name and path should be close to transactionRequest.

Reply:
This looks like a generic CSV Writer. What makes you say that this is TxnRequest specific?

Reply:
`CsvUtility` is the assembly line for `CsvWriter<T>`.
rit now since only `transactionRequest` is using `CsvUtility` it seems that it is specific but it is generic. 

CSV operations  like writing headers, and writing rows are more of low-level CSV handling which is taken care of in the generic class `CsvWriter<T>`.

And consuming of `CsvWriter<T>` is done in `CsvUtility`, tasks like getting fields for respective POJOs, and then generating headers from it, and then building the `CsvWriter<T>` object etc. 

I have chosen this design because of the following reason.
1. Routes don't need to know how to write to CSV, hence `CsvUtility` will do it for you. A single source of truth for writing to csv.
2. Csv writing includes many generic steps which will be common for all the DTOs, hence generic abstraction using class `CsvWriter<T>`.
3. Later we can extend the `CsvWriter<T>` support for multiple things like file generation, HTTP stream wiritng etc, this won't disturb the existing use case.

Reply:
`CSV writer`  is generic and the design is impressive. But the `CSV Utility` is a wrapper and since it is using the `transactionRequest ` I felt that is use case-specific. In the future, we have to write more static methods in the `Csv Utility` to serve multiple use cases. Rather than that, we can make multiple wrappers(Utility) for `CSV writer`, and each will be specific to that use case(Single Responsibility Principle).  
It will make sure that the implementation detail (Assembly line)is done within each scope, not in the same 'Csv Utility' class.

Reply:
Ok, then what about common helper methods? won't there be code redundancy? For example
`CsvUtility` rit now has one method called [String[] getCsvHeader(String[] fields)](https://github.com/danishjamal104/ph-ee-operations-app/blob/d525c484b3a2beba8036e9a4db8e4f39eecc800c/src/main/java/org/apache/fineract/utils/CsvUtility.java#L44) this is a generic helper method to generate the header, this method can't be in respective scope since it is generic for all and neither can't it be in `CSV writer`

Reply:
@danishjamal104  But what about `transactionRequestsFields` and `writeToCsv` methods, they are scope specific.  ` get CSV header` can be moved to `csvwriter`.   

 Actually, in your case, the `getCsvHeader` method may not even be needed, coz you are just looping and formatting a list of data that is hardcoded.  If you just hardcode the data in the format you require( Upper case here) why do you need  `getCsvHeader` ?   I expect that this method can have a future scope when headers are not hardcoded, if that is the case, you can keep it at a commonplace.

Reply:
@fynmanoj now CsvUtility is also generic, see the updated implementation.